test/schema.test.ts

```diff
@@ -186,6 +175,170 @@ describe('schema', () => {
+      const numYears = 5;
+      const cardinalityData = [];
+      for (var i = 1; i <= numYears; i++) {
+        cardinalityData.push({
+          a: 'June ' + i + ', 200' + i
+        });
+      }
+      const cardinalitySchema = Schema.build(cardinalityData);
+      const cardinality: number = cardinalitySchema.cardinality({
+        field: 'a',
+        channel: Channel.X,
+        timeUnit: 'yearmonth'
+      });
+      assert.equal(cardinalitySchema.primitiveType('a'), PrimitiveType.DATE);
+      assert.equal(cardinality, numYears);
+    });
+
+    it('should correctly compute cardinality for `yearquarter` timeunit', () => {
+      let cardinalityData = [{a: 'June 21, 1996'}, {a: 'January 21, 1996'}];
+      let cardinalitySchema = Schema.build(cardinalityData);
+      let cardinality: number = cardinalitySchema.cardinality({
+        field: 'a',
+        channel: Channel.X,
+        timeUnit: 'yearquarter'
+      });
+      assert.equal(cardinalitySchema.primitiveType('a'), PrimitiveType.DATE);
+      assert.equal(cardinality, 2);
+
+      cardinalityData = [{a: 'June 21, 1996'}, {a: 'May 21, 1996'}];
+      cardinalitySchema = Schema.build(cardinalityData);
+      cardinality = cardinalitySchema.cardinality({
+        field: 'a',
+        channel: Channel.X,
+        timeUnit: 'yearquarter'
+      });
+      assert.equal(cardinalitySchema.primitiveType('a'), PrimitiveType.DATE);
+      assert.equal(cardinality, 1);
+    });
+
+    it('should correctly compute cardinality for `year` timeunit', () => {
+      let cardinalityData = [{a: 'June 21, 2000'}, {a: 'June 21, 2000'}];
+      let cardinalitySchema = Schema.build(cardinalityData);
+      let cardinality: number = cardinalitySchema.cardinality({
+        field: 'a',
+        channel: Channel.X,
+        timeUnit: 'year'
+      });
+      assert.equal(cardinalitySchema.primitiveType('a'), PrimitiveType.DATE);
+      assert.equal(cardinality, 1);
+
+      cardinalityData = [{a: 'June 21, 2000'}, {a: 'June 21, 2010'}];
+      cardinalitySchema = Schema.build(cardinalityData);
+      cardinality = cardinalitySchema.cardinality({
+        field: 'a',
+        channel: Channel.X,
+        timeUnit: 'year'
+      });
+      assert.equal(cardinalitySchema.primitiveType('a'), PrimitiveType.DATE);
+      assert.equal(cardinality, 2);
+    });
+
+    it('should correctly compute cardinality for `quartermonth` timeunit', () => {
+      // should be the same as the 'month' cardinality
+      let cardinalityData = [
+        {a: 'June 1, 2000'},
+        {a: 'May  1, 2000'},
+        {a: 'May  1, 2000'},
+        {a: 'January  1, 2000'}
+      ];
+      let cardinalitySchema = Schema.build(cardinalityData);
+      let cardinality: number = cardinalitySchema.cardinality({
+        field: 'a',
+        channel: Channel.X,
+        timeUnit: 'quartermonth'
+      });
+      assert.equal(cardinalitySchema.primitiveType('a'), PrimitiveType.DATE);
+      assert.equal(cardinality, 3);
```

Comment:
I believe this is correct because there are three different months, and month should "override" quarter? So the cardinality is 3?